test/SparqlAntlrParserTest.cpp

```diff
@@ -104,29 +106,24 @@ using ::testing::Pair;
 using ::testing::SizeIs;
 using ::testing::StrEq;
 
-namespace {
-template <typename Exception = ParseException>
-void expectNumericLiteralFails(const string& input) {
-  EXPECT_THROW(parseNumericLiteral(input), Exception) << input;
-}
-}  // namespace
-
 TEST(SparqlParser, NumericLiterals) {
-  testNumericLiteral("3.0", 3.0);
-  testNumericLiteral("3.0e2", 300.0);
-  testNumericLiteral("3.0e-2", 0.030);
-  testNumericLiteral("3", (int64_t)3ll);
-  testNumericLiteral("-3.0", -3.0);
-  testNumericLiteral("-3", (int64_t)-3ll);
-  testNumericLiteral("+3", (int64_t)3ll);
-  testNumericLiteral("+3.02", 3.02);
-  testNumericLiteral("+3.1234e12", 3123400000000.0);
-  testNumericLiteral(".234", 0.234);
-  testNumericLiteral("+.0123", 0.0123);
-  testNumericLiteral("-.5123", -0.5123);
-  testNumericLiteral(".234e4", 2340.0);
-  testNumericLiteral("+.0123E-3", 0.0000123);
-  testNumericLiteral("-.5123E12", -512300000000.0);
+  auto expectNumericLiteral = ExpectCompleteParse<&Parser::numericLiteral>{};
+  auto expectNumericLiteralFails = ExpectParseFails<&Parser::numericLiteral>{};
+  expectNumericLiteral("3.0", m::NumericLiteralFP(3.0));
+  expectNumericLiteral("3.0e2", m::NumericLiteralFP(300.0));
+  expectNumericLiteral("3.0e-2", m::NumericLiteralFP(0.030));
+  expectNumericLiteral("3", m::NumericLiteralWhole(3ll));
+  expectNumericLiteral("-3.0", m::NumericLiteralFP(-3.0));
+  expectNumericLiteral("-3", m::NumericLiteralWhole(-3ll));
+  expectNumericLiteral("+3", m::NumericLiteralWhole(3ll));
```

Comment:
You were talking about making the `EQ` default matcher configurabel, s.t we can easily make this
```
auto expectFloat = exepectCompleteParse<&parseNumericLiteral, m::NumericLiteralFP>{};
expectFloat("3.0", 3.0)
```
Maybe we can talk about this in a meeting how it can be done, you basically need to make the `Value` a template that
`requires` that it is a valid argument to the `DefaultMatcher`.